l10n_us_check_writing_address/report/print_check.py

```diff
@@ -0,0 +1,135 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 Ursa Information Systems <http://www.ursainfosystems.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+
+from odoo.osv import osv
+from odoo.report import report_sxw
+from odoo import api, fields, models, _
+
+LINE_FILLER = '*'
+INV_LINES_PER_STUB = 9
+
+
+class ReportPrintCheck(report_sxw.rml_parse):
+
+    def __init__(self, cr, uid, name, context):
+        super(report_print_check, self).__init__(cr, uid, name, context)
+        self.localcontext.update({
+            'pages': self.get_pages,
+        })
+
+    def fill_line(self, amount_str):
+        return amount_str and (amount_str+' ').ljust(200, LINE_FILLER) or ''
+
+    def get_pages(self, payment):
+        """ Returns the data structure used by the template : a list of dicts containing what to print on pages.
+        """
+        stub_pages = self.make_stub_pages(payment)
+        multi_stub = payment.company_id.us_check_multi_stub
+        pages = []
+        for i in range(0, stub_pages != None and len(stub_pages) or 1):
+            pages.append({
+                'sequence_number': payment.check_number\
+                    if (payment.journal_id.check_manual_sequencing and payment.check_number != 0)\
+                    else False,
+                'payment_date': payment.payment_date,
+                'partner_name': payment.partner_id.name,
+                'partner_name_addr': payment.partner_id.street,
+                'partner_name_addr2': payment.partner_id.street2,
+                'partner_city': payment.partner_id.city,
+                'partner_state': payment.partner_id.state_id.code,
+                'partner_zip': payment.partner_id.zip,
+                'currency': payment.currency_id,
+                'amount': payment.amount if i == 0 else 'VOID',
```

Comment:
This `i == 0` is being performed multiple times. It would be more efficient and concise to perform once above, assigning to temp vars for use in the dict assignment.